### PR TITLE
Cumulative Layout Shift improvements

### PR DESCRIPTION
**Pull request by ntorbinskiy**
Cls optimisation

### CODE REVIEW

File: components/PostCard.tsx [hunk 1]
```diff
@@ -1,4 +1,5 @@
 import { PostDocumentWithoutContent } from "interfaces";
+// import Image from "next/future/image";
```
Comment (1alexvash) [inline components/PostCard.tsx:2]: Delete this line, if it's not used

File: config/index.ts [hunk 2]
```diff
@@ -1,19 +1,19 @@
...
+	author_position: "Full Stack JavaScript Developer",
+	social_links: [
+		{ link: "https://t.me/HeWorksSoHard", image: "/images/telegram.svg", width: 20.99, height: 16.99 },
+		{ link: "https://github.com/1alexvash", image: "/images/github-gray.svg", width: 19, height: 19 },
+		{ link: "mailto:1alexvash@gmail.com", image: "/images/mail.svg", width: 24, height: 15.99 },
+	],
+	host_url: "https://alex-code.com/",
+	defaultImage:
+		"https://alex-code.com/post-images/2022-08-05-book-review-so-good-they-cant-ignore-you.webp",
+	posts_per_page: 12,
```
Comment (1alexvash) [inline config/index.ts:16]: @ntorbinskiy file is still not formatted
animation

It gets annoying, to review the same thing over and over again for 5 times

Comment (ntorbinskiy) [reply config/index.ts]: I'm sorry, there was  accidently turned on another formatter.  I've fixed it in the next commit.

**Review by 1alexvash — CHANGES_REQUESTED** [review]

File: components/Footer.tsx [hunk 3]
```diff
@@ -7,13 +7,23 @@ const Footer = () => (
     <div className="container">
       <div className="footer-content">
         <div className="footer-logo">
-          <img src="/images/footer-logo.svg" alt="footer-logo" />
+          <img
+            src="/images/footer-logo.svg"
+            alt="footer-logo"
+            width={180}
+            height={50.99}
```
Comment (1alexvash) [inline components/Footer.tsx:14]: @ntorbinskiy 
image

How do you even get those decimal points?
Can you remove them all?

File: components/Header/Logo.tsx [hunk 4]
```diff
@@ -5,8 +5,20 @@ import React from "react";
...
   const logo = (
     <a className="main-logo">
-      <img className="light" src="/images/main-logo.svg" alt="light-theme" />
-      <img className="dark" src="/images/main-logo-dark.svg" alt="dark-theme" />
+      <img
+        className="light"
+        src="/images/main-logo.svg"
+        alt="light-theme"
+        width={180}
+        height={50.99}
```
Comment (1alexvash) [inline components/Header/Logo.tsx:13]: why 50.99 instead of 51?

File: components/Header/Logo.tsx [hunk 5]
```diff
@@ -5,8 +5,20 @@ import React from "react";
...
+      />
+      <img
+        className="dark"
+        src="/images/main-logo-dark.svg"
+        alt="dark-theme"
+        width={180}
+        height={50.99}
```
Comment (1alexvash) [inline components/Header/Logo.tsx:20]: same story

File: components/Header/ThemeSwitcher.tsx [hunk 6]
```diff
@@ -32,10 +32,10 @@ const ThemeSwitcher = () => {
   return (
     <ul className="theme-switcher">
       <li className="light" onClick={() => setTheme("light")}>
-        <img src="/images/sun.svg" alt="sun" />
+        <img src="/images/sun.svg" alt="sun" width={20} height={20} />
       </li>
       <li className="dark" onClick={() => setTheme("dark")}>
-        <img src="/images/moon.svg" alt="moon" />
+        <img src="/images/moon.svg" alt="moon" width={19} height={17.99} />
```
Comment (1alexvash) [inline components/Header/ThemeSwitcher.tsx:38]: Remove numbers after decimal point

File: components/Post/LatestPosts.tsx [hunk 7]
```diff
@@ -17,7 +17,12 @@ const LatestPosts = ({ latestPosts }: Props) => {
         <div className="related-posts-block" key={post.title}>
           <Link href={post.slug}>
             <a className="image">
-              <img src={post.featuredImage} alt="blog post image" />
+              <img
+                src={post.featuredImage}
+                alt="blog post image"
+                width={101.99}
+                height={101.99}
```
Comment (1alexvash) [inline components/Post/LatestPosts.tsx:24]: ?

File: config/index.ts [hunk 8]
```diff
@@ -6,9 +6,24 @@ const config = {
...
+      image: "/images/github-gray.svg",
+      width: 19,
+      height: 19,
+    },
+    {
+      link: "mailto:1alexvash@gmail.com",
+      image: "/images/mail.svg",
+      width: 24,
+      height: 15.99,
+    },
```
Comment (1alexvash) [inline config/index.ts:26]: Here as well

File: pages/index.tsx [hunk 9]
```diff
@@ -38,13 +38,13 @@ const Home: NextPage<{ posts: PostDocumentWithoutContent[] }> = ({ posts }) => {
...
       <Head>
         <title>{config.site_title}</title>
-        <meta name="description" content={config.site_description} />
         <meta property="og:title" content={config.site_keywords[1]} />
         <meta property="og:description" content={config.site_description} />
         <meta property="og:url" content={config.host_url} />
         <meta property="og:type" content="website" />
         <meta property="og:site_name" content={config.site_title} />
         <meta property="og:image" content={config.defaultImage} />
+        <meta name="description" content={config.site_description} />
```
Comment (1alexvash) [inline pages/index.tsx:47]: @ntorbinskiy this branch, is not about SEO, does it belong in here?

**Review by 1alexvash — CHANGES_REQUESTED** [review]

File: components/Footer.tsx (hunk 3 above)
Comment (ntorbinskiy) [reply components/Footer.tsx:14]: I did incorrect calculations, sorry. In the next commit I'll fix these mistakes

File: pages/index.tsx (hunk 9 above)
Comment (ntorbinskiy) [reply pages/index.tsx:47]: No. When I was doing SEO optimization, I've got a problem with 2 titles of posts, So, I checked it in this branch and then removed. Now I understand,  that I needed to create another branch for it

File: styles/main.css [hunk 10]
```diff
@@ -778,8 +777,28 @@ header {
...
+	top: 0;
+	height: 100%;
+	width: 100%;
+	z-index: 5;
+	border: 1px solid transparent;
+	border-radius: 3px;
+	position: absolute;
+	display: block;
+	transition: all 0.3s ease-out;
+	pointer-events: none;
```
[the remark this answers is not in the document]
Comment (ntorbinskiy) [reply styles/main.css:799]: I can format the whole project by command in terminal in one commit. Do you agree?

File: pages/index.tsx (hunk 9 above)
Comment (ntorbinskiy) [reply pages/index.tsx:47]: Should I turn this block back?

File: styles/main.css (hunk 10 above)
[the remark this answers is not in the document]
Comment (1alexvash) [reply styles/main.css:799]: @ntorbinskiy what command?

File: pages/index.tsx (hunk 9 above)
Comment (1alexvash) [reply pages/index.tsx:47]: Yeah, probbably remove this code in this branch, as it is not relevant

File: styles/main.css (hunk 10 above)
[the remark this answers is not in the document]
Comment (ntorbinskiy) [reply styles/main.css:799]: > @ntorbinskiy what command?

npx prettier --write .

File: pages/index.tsx (hunk 9 above)
Comment (ntorbinskiy) [reply pages/index.tsx:47]: 👌

File: styles/main.css (hunk 10 above)
[the remark this answers is not in the document]
Comment (1alexvash) [reply styles/main.css:799]: @ntorbinskiy awesome

[the remark this answers is not in the document]
Comment (1alexvash) [reply styles/main.css:799]: image
push changes, so I can see it as well

**Comment by 1alexvash** [comment]
@ntorbinskiy 

Almost done with this branch:

Things left:
1) Send files in updated formatting
2) Take a look at this hover effect thing, there's definitly a way to make it a couple of lines shorter by ussing CSS coma selector for example, or just by applying this effect for both theme initially
3) Resolve merge conflict with header file

**Comment by ntorbinskiy** [comment]
Resolved merge conflicts

**Comment by ntorbinskiy** [comment]
image
image
image
image
image
Это твой код? Просто я его не писал. это components/Header/index.tsx

File: config/index.ts (hunk 2 above)
Comment (1alexvash) [reply config/index.ts]: @ntorbinskiy maybe this other formatter is causing problems with break lines on the SEO branch?

File: components/Post/PostContent.tsx [hunk 11]
```diff
@@ -31,7 +31,12 @@ const PostContent = ({ post }: Props) => {
       </Head>
 
       <div className="blogpost-image">
-        <img src={post.featuredImage} alt="blog post image" />
+        <img
+          src={post.featuredImage}
+          alt="blog post image"
+          width={790}
+          height={300}
```
Comment (1alexvash) [inline components/Post/PostContent.tsx:38]:
```suggestion
          height={394}
```

File: styles/main.css (hunk 10 above)
[the remark this answers is not in the document]
Comment (1alexvash) [reply styles/main.css]: @ntorbinskiy 
image
This code is not needed, as white theme code does the work for the dark-theme stuff

**Pull request merged**